experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/NodeManagerCallbackHandler.scala

```diff
@@ -39,4 +40,15 @@ class NodeManagerCallbackHandler(am: ActorRef) extends NMClientAsync.CallbackHan
   def onStopContainerError(containerId: ContainerId, throwable: Throwable) {
     LOG.error(s"Container exception : $containerId", throwable)
   }
-}
\ No newline at end of file
+}
+
+trait NodeManagerCallbackHandlerFactory {
```

Comment:
Instead of declaring a factory, in scala, there is a cleaner way to do dependency injection, Like function itself.

For example. instead of 

```scala
class AmActor(...
              nodeManagerCallbackHandlerFactory: NodeManagerCallbackHandlerFactory) extends Actor {

```
You can pass function like

```scala
class AmActor(...
              nodeManagerCallbackHandlerFactory: () => CallbackHandler) extends Actor {
```

Then you can avoid declaring boilplate factory class.